backend/api/src/main/kotlin/software/shonk/application/service/ShorkService.kt

```diff
@@ -70,6 +69,26 @@ class ShorkService(private val shork: IShork) : ShorkUseCase {
         return Result.success(lobby.getStatus())
     }
 
+    override fun getAllLobbies(): List<LobbyStatus> {
+        return lobbies.keys.mapNotNull { lobbyId ->
+            getLobbyStatus(lobbyId).getOrNull()?.let { status ->
+                LobbyStatus(
+                    lobbyId = lobbyId,
+                    playersJoined =
+                        listOfNotNull(
+                            if (lobbies[lobbyId]?.programs?.containsKey("playerA") == true)
+                                "playerA"
+                            else null,
+                            if (lobbies[lobbyId]?.programs?.containsKey("playerB") == true)
+                                "playerB"
+                            else null,
+                        ),
```

Comment:
While this works, it will break in the feature once/if more than just those two player(s / player names) are allowed.
Why do you not just use the keys in the map directly?

Reply:
you're right about this. I initially focused on hardcoding player names to address the current requirements, but I now see the value in making the solution dynamic for future scalability